crates/red_knot_project/src/metadata/options.rs

```diff
@@ -385,12 +385,12 @@ impl Diagnostic for OptionDiagnostic {
         Cow::Borrowed(&self.message)
     }
 
-    fn file(&self) -> Option<File> {
-        self.file
-    }
-
-    fn range(&self) -> Option<TextRange> {
-        self.range
+    fn span(&self) -> Option<Span> {
```

Comment:
Could we directly store the `Span` on `OptionDiagnostic` instead of constructing it here on the fly?

Reply:
This is done.